### ISSUE TITLE
Http Basic Auth 2 Factor Auth with Duo

### ISSUE THREAD

**Issue by CreatorSimon** [authelia/authelia#4069]
### Description

Currently the HTTP Basic Auth configuration only works with One Factor, which is fine but a possibility to use 2 Factor Auth with DUO would be awesome!

### Use Case

Use Case: Gitlab or any other GIT application server with public access. You can use HTTP Basic auth for git cli commands but no DUO 2 Factor auth, which would be awesome to get extra protection.

**Comment by james-d-elliott** [comment]
Describe the user experience in your use cases. How would a user know to use a 2FA method? How would they enter the details?

**Comment by CreatorSimon** [comment]
User experience: So we have an Gitlab installation which is public available trough a reverse proxy. We are using Authelia as an Application Firewall to protect our public accessible services/applications. The problem is, that we want access to the repositories via 2 Factor Auth. With the Authelia redirecting flow (recommended flow), it isnt possible to access the repos. But with HTTP Basic Auth it is possible to access the repos. 

For example: "git clone https://our.gitlab.com/project/project.git"
Would ask for username and password which works with HTTP Basic Auth. But then a 2nd Factor authentication with DUO would be awesome to make it more secure. 

Hopefully it's more understandable now. Thanks for this awesome project!

**Comment by james-d-elliott** [comment]
I don't believe this is even possible. There is no web standard which covers 2FA in this manner that I'm aware of, and git would have to support it if one existed. That or it would have to support sessions which I don't believe it does. I'm happy to be corrected however.

**Comment by CreatorSimon** [comment]
We currently use an Apache2 reverse proxy setup which includes http Basic auth (ldap) with duo. I dont know the exact config so i can't reengineer the flow right now, maybe i can figure it out. But it works perfectly with the given use case. So thats the reason why I proposed the feature. 

For example:
git clone https://gitlab.com/project/project.git 
>Username: ldap.superuser
>Password: superpassword
---If the ldap creds are correct we get an duo push, if we accept it the 2FA is completed and we can access the git---
*Just for example*

I know its very special, but we want to use this awesome project as an easy solution to protect all our applications.

**Comment by james-d-elliott** [comment]
This is probably how it's done: https://blog.neilsabol.site/post/duo-push-mfa-2-factor-authentication-for-apache-httpd/

Seems to use a ldap proxy server that requires Duo before the credentials are confirmed. The equivalent if Authelia did it would delay ALL authentication attempts with Duo.

**Comment by james-d-elliott** [comment]
Thanks for submitting this feature request, we greatly appreciate ideas like this from the community. We've decided to for the time being convert this issue to an idea discussion for the following reasons:

- This prevents our issue list being obscure due to excessive items that we have no direct plans to implement at this stage.
- This prevents this reasonable idea from being lost if it were closed.
- This allows people to vote on the idea.
